### ISSUE TITLE
Symlink support

### ISSUE THREAD

**Issue by aemino** [standard-things/esm#21]
Hi. In one of my current projects, I have Project A and Project B. Project B is a dependency of Project A. In the interest of developing both simultaneously, I have Project B as a symlinked dependency in the `node_modules` of Project A. As I was testing this awesome package, I noticed that it seemingly doesn't support symlinks. When I try to require Project B from Project A, an error is thrown claiming it can't find Project B.

**Comment by jdalton** [comment]
@aemino Could you set up a simple repro repo. Internally we should be using `Module._resolveFilename()` which should follow symlinks.

**Comment by aemino** [comment]
@jdalton I created a test repro [here](https://github.com/aemino/esm-repro). Let me know if the real issue is me doing something incorrectly. 😅

**Comment by aemino** [comment]
I'd like to add a follow up to this. After additional testing, it appears that this issue is my fault. My working environment involves using WSL, which creates Unix-esque symlinks, while VSCode (what I'm using to run the code with) uses the Windows built-in command line in the debug console. The different terminals create different types of symlinks that are not compatible.

Thank you for taking the time to respond.

**Comment by jdalton** [comment]
@aemino No worries. Thanks for following up.

Feel free to ping back if you find something else too :D

**Comment by aemino** [comment]
I won't create a new issue for this as it may very well just be that I'm missing something blatant @jdalton, but here goes.

For some reason, I can't seem to get exports working. I've tested on Node 8.3 (Windows), Node 8.3 (Linux), and Node 7.10, Linux).

Here's the example code I was using.
```js
// index.js
require('@std/esm');
console.log(require('./index.mjs'));

// index.mjs
export const foo = 'bar';
```
Running `index.js` just resulted in the output `{}`. It seems as though this is working for everyone else (and the tests), but for some reason not for me. Please let me know if I'm doing something horribly wrong. Thanks! 😄

**Comment by jdalton** [comment]
Thanks again @aemino!

I'm looking into it. I believe it's related to the second half of https://github.com/standard-things/esm/issues/15#issuecomment-321843670.

**Comment by aemino** [comment]
@jdalton Thanks for your responses and promptness. I look forward to the fix! 😃

**Comment by jdalton** [comment]
Fixed with v0.3.0 🎉
